source/slang/slang-type-system-shared.h

```diff
@@ -58,6 +58,20 @@ FOREACH_BASE_TYPE(DEFINE_BASE_TYPE)
     const int kStdlibTextureIsShadowParameterIndex = 6;
     const int kStdlibTextureIsCombinedParameterIndex = 7;
     const int kStdlibTextureFormatParameterIndex = 8;
+
+    enum class AddressSpace : uint64_t
+    {
+        Generic = 0x7fffffff,
+        ThreadLocal = 1,
```

Comment:
Actually I suggest we can define the enum by using the storage class directly,
e.g.
`ThreadLocal = (uint64_t)(SpvStorageClassInput),`

so we won't be impacted if there is any change of the spirv.h, though it's very unlikely to change.

Reply:
`ThreadLocal = (uint64_t)(SpvStorageClassInput),` would be invalid semantics, these are not actually equivalents.

SPIRV uses `SpvStorageClassFunction`(7) while other targets use `ThreadLocal`(1).

Reply:
There is no one to one mapping from spv storage class to what is currently defined in `AddressSpace`. The current AddressSpace enum is derived from what metal requires and did not consider the need of SPIRV. For example, ThreadLocal can be SpvFunction or SpvPrivate depending on whether it is a local or global variable.

This will need a future pass for a full conversion so we never directly use the enums for both spirv and metal. For now we just use the SpvStorageClass values directly as AddressSpace.